File: 55 Jump Game/Java/Solution.java
public class Solution {
    public boolean canJump(int[] nums) {
        int max = 0;
        int i = 0;
        while (i < nums.length - 1) {
            if (max == i && nums[i] == 0) {
                return false;
            }
            max = Math.max(max, i + nums[i]);
            i++;
        }
        return max >= i;
    }
}

File: 55 Jump Game/C/solution.c
#include "solution.h"
#include "math.h"

bool canJump(int* nums, int numsSize) {
    int m = 0;
    int i = 0;
    while (i < numsSize - 1) {
        if (m == i && nums[i] == 0) {
            return false;
        }
        m = max(m, i + nums[i]);
        i++;
    }
    return m >= i;
}